tests/drop_counters/fanout/mellanox/mellanox_fanout.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from ..fanout_base import BaseFanoutHandler
+
+MAX_OPENFLOW_RULE_ID = 65535
+
+class FanoutHandler(BaseFanoutHandler):
+    def __init__(self, testbed_devices):
+        self.initialized = False
+        self.rule_id = MAX_OPENFLOW_RULE_ID
+        self.run_ansible_cli_cmd = "cd {ansible_path}; ansible-playbook {playbook} -i lab -l {fanout_host} --extra-vars \"{extra_vars}\" -vvvvv"
+        # Ansible localhost fixture which calls ansible playbook on the local host
+        self.ansible_localhost = testbed_devices["localhost"]
+        # Ansible playbook which executes Jinja template
+        self.ansible_playbook = os.path.join(os.path.dirname(__file__), "exec_template.yml")
+        # Jinja template which delete specific openflow rule
+        self.del_rule_template = os.path.join(os.path.dirname(__file__), "mlnx_del_of_rule.j2")
+        # Ansible config files
+        self.lab_conn_graph_file = os.path.normpath((os.path.join(os.path.dirname(__file__), "../../../../ansible/files/lab_connection_graph.xml")))
+        self.ansible_root = os.path.normpath((os.path.join(__file__, "../../../../../ansible")))
+
+        dut_facts = self.ansible_localhost.conn_graph_facts(host=testbed_devices["dut"].hostname, filename=self.lab_conn_graph_file)["ansible_facts"]
+        self.fanout_host = dut_facts["device_conn"]["Ethernet0"]["peerdevice"]
+        fanout_facts = self.ansible_localhost.conn_graph_facts(host=self.fanout_host, filename=self.lab_conn_graph_file)["ansible_facts"]
+
+        self.fanout_trunk_port = None
+        for iface, iface_info in fanout_facts["device_port_vlans"].items():
+            if iface_info["mode"] == "Trunk":
+                self.fanout_trunk_port = iface[iface.find("/") + 1:]
+                break
+        else:
+            raise Exception("Unable to identify Fanout Trunk port")
+
+    def update_config(self, **kwargs):
+        """
+        Execute Jinja template on fanout switch which creates openflow rule.
+        @kwargs: Parameters used as 'extra-vars' for 'ansible-playbook' CLI utility.
+        """
+        attempt = 100
+        while attempt > 0:
```

Comment:
Any reason we can't use a for loop here?

Reply:
Fixed.